src/overlay/Peer.cpp

```diff
@@ -743,10 +750,22 @@ Peer::recvTransaction(StellarMessage const& msg)
             // record that this peer sent us this transaction
             mApp.getOverlayManager().recvFloodedMsg(msg, shared_from_this());
 
+            size_t size = xdr::xdr_argpack_size(msg);
             if (recvRes == TransactionQueue::AddResult::ADD_STATUS_PENDING)
             {
                 // if it's a new transaction, broadcast it
                 mApp.getOverlayManager().broadcastMessage(msg);
+
+                mUniqueBytesRecv.Mark(size);
+            }
+            else
+            {
+                mDuplicateBytesRecv.Mark(size);
+                CLOG(TRACE, "Perf")
```

Comment:
As a general thing (we need to be better about this) looking up CLOG(TRACE, ...) is nontrivial cost; it's worth caching trace-being-disabled (which it usually is) or hoisting that out of a thing that's going to be called thousands of times a second.